pubsub/api/Pubsub.Samples.Tests/PullMessagesAsyncWithDeliveryAttemptsTest.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Xunit;
+
+[Collection(nameof(PubsubFixture))]
+public class PullMessagesAsyncWithDeliveryAttemptsTest
+{
+    private readonly PubsubFixture _pubsubFixture;
+    private readonly PublishMessagesAsyncSample _publishMessagesAsyncSample;
+    private readonly PullMessagesAsyncWithDeliveryAttemptsSample _pullMessagesAsyncWithDeliveryAttemptsSample;
+    private readonly CreateSubscriptionWithDeadLetterPolicySample _createSubscriptionWithDeadLetterPolicySample;
+
+    public PullMessagesAsyncWithDeliveryAttemptsTest(PubsubFixture pubsubFixture)
+    {
+        _pubsubFixture = pubsubFixture;
+        _publishMessagesAsyncSample = new PublishMessagesAsyncSample();
+        _pullMessagesAsyncWithDeliveryAttemptsSample = new PullMessagesAsyncWithDeliveryAttemptsSample();
+        _createSubscriptionWithDeadLetterPolicySample = new CreateSubscriptionWithDeadLetterPolicySample();
+    }
+
+    [Fact]
+    public async void PullMessagesAsyncWithDeliveryAttempts()
+    {
+        string randomName = _pubsubFixture.RandomName();
+        string topicId = $"testTopicForDeadLetterPolicyMessageSyncAck{randomName}";
+        string subscriptionId = $"testSubscriptionDeadLetterPolicyForMessageSyncAck{randomName}";
+        var message = _pubsubFixture.RandomName();
+
+        _pubsubFixture.CreateTopic(topicId);
+        _createSubscriptionWithDeadLetterPolicySample.CreateSubscriptionWithDeadLetterPolicy(
+           _pubsubFixture.ProjectId, subscriptionId, topicId, _pubsubFixture.DeadLetterTopic);
+
+        _pubsubFixture.TempSubscriptionIds.Add(subscriptionId);
+
+        await _publishMessagesAsyncSample.PublishMessagesAsync(_pubsubFixture.ProjectId, topicId, new string[] { message });
+
+        // Pull and acknowledge the messages
+        var deliveryAttempt = await _pullMessagesAsyncWithDeliveryAttemptsSample.PullMessagesAsyncWithDeliveryAttempts(_pubsubFixture.ProjectId, subscriptionId, false);
+        Assert.True(deliveryAttempt <= 1);
```

Comment:
It's now failing here, because we are running for 5 seconds, during which the message can be attempted several times because we don't ack.
So I believe the asserts are interchanged, here it should be:
```suggestion
        Assert.True(deliveryAttempt >= 1);
```

Reply:
Delivery attempt is best effort only. As long as this field is not empty, we shall let tests pass. https://cloud.google.com/pubsub/docs/dead-letter-topics#adding_dead-letter_topics

Reply:
OK, so maybe instead of making 2 calls one for nack and one for ack, let's keep it to one for ack and test that deliveryAttempt > 0? Does that sound right?

Reply:
Sounds good! Yeah, it would give samples a more robust test (which I like!) but this feature should have already been tested in the library, so we've been keeping samples simple (in other languages too).

Reply:
fix it with only one call with ack and tested deliveryAttempt > 0.
increase delay time in sample as sample test was flaky.